js/bitstamp.js

```diff
@@ -1140,36 +1140,73 @@ module.exports = class bitstamp extends Exchange {
-            'deposit': {},
-        };
-    }
-
     async fetchTransactionFees (codes = undefined, params = {}) {
         /**
          * @method
          * @name bitstamp#fetchTransactionFees
          * @description fetch transaction fees
-         * @param {[string]|undefined} codes not used by bitstamp fetchTransactionFees ()
+         * @see https://www.bitstamp.net/api/#balance
+         * @param {[string]|undefined} codes list of unified currency codes
          * @param {object} params extra parameters specific to the bitstamp api endpoint
          * @returns {[object]} a list of [fee structures]{@link https://docs.ccxt.com/en/latest/manual.html#fee-structure}
          */
         await this.loadMarkets ();
-        const balance = await this.privatePostBalance (params);
-        return this.parseTransactionFees (balance);
+        const response = await this.privatePostBalance (params);
+        //
+        //  {
+        //     yfi_available: '0.00000000',
+        //     yfi_balance: '0.00000000',
+        //     yfi_reserved: '0.00000000',
+        //     yfi_withdrawal_fee: '0.00070000',
+        //     yfieur_fee: '0.000',
+        //     yfiusd_fee: '0.000',
+        //     zrx_available: '0.00000000',
+        //     zrx_balance: '0.00000000',
+        //     zrx_reserved: '0.00000000',
+        //     zrx_withdrawal_fee: '12.00000000',
+        //     zrxeur_fee: '0.000',
+        //     zrxusd_fee: '0.000',
+        //     ...
+        //  }
+        //
+        const result = {};
+        let infoObject = {};
+        let infoObjectLen = undefined;
+        let prevCode = undefined;
+        let mainCurrencyId = undefined;
+        const ids = Object.keys (response);
+        for (let i = 0; i < ids.length; i++) {
+            const id = ids[i];
+            const currencyId = id.split ('_')[0];
+            const code = this.safeCurrencyCode (currencyId);
+            if (codes !== undefined && !this.inArray (code, codes)) {
+                continue;
+            }
+            if (id.indexOf ('_available') >= 0) {
+                mainCurrencyId = currencyId;
+                result[code] = {
+                    'deposit': undefined,
+                    'withdraw': undefined,
+                    'info': undefined,
+                };
+                infoObjectLen = infoObject.length;
+                if (infoObjectLen > 0) {
+                    result[prevCode]['info'] = infoObject;
+                    infoObject = {};
+                }
+            }
+            if (currencyId === mainCurrencyId) {
+                infoObject[id] = this.safeNumber (response, id);
+            }
+            if (id.indexOf ('_withdrawal_fee') >= 0) {
+                result[code]['withdraw'] = this.safeNumber (response, id);
+            }
+            prevCode = this.safeCurrencyCode (mainCurrencyId);
+            if ((i === ids.length - 1 && infoObjectLen > 0) || this.inArray (code, codes)) {
```

Comment:
@AlgoQ this crashes when the user does not provide `codes` 
![image](https://user-images.githubusercontent.com/43336371/199475941-5e4929db-e01d-42b5-aecd-eae32258b4de.png)